### ISSUE TITLE
New video grid not appending when a new user comes in

### ISSUE THREAD

**Issue by DavidArmendariz** [itstaranarora/video-chat-v1#2]
Hi! When a new user joins the room, it looks like the "join-room" event is not emitted.
I put a `console.log` on:

```
peer.on("open", (id) => {
  socket.emit("join-room", ROOM_ID, id, user);
});
```

And it does not reach this part. Thus, the `user-connected` event does not get broadcasted.

**Comment by BhaskarReddy7** [comment]
Same here only one user are able to see

**Comment by mccombs** [comment]
Following. Same Issue. :(

**Comment by aastha271100** [comment]
Did this issue get resolved?

**Comment by itstaranarora** [comment]
This problem is due to STUN server. This app use STUN server which does not
work with all the firewall / NAT setups. so you may need to switch to a
TURN server.

**Comment by aastha271100** [comment]
Is the send/recieve functionality of messages not working due to this only?

**Comment by itstaranarora** [comment]
Yes I think so. You can research on this thing this is what I found when I
was trying to fix this errror

**Comment by DoctorQuasi** [comment]
if you refresh the browser, the webcam windows appear.

**Comment by rajibhalder** [comment]
I have issue with below piece of code,
**_video.addEventListener('loadedmetadata', () => {
      video.play();
    });_**
If my two socketio client are at different network then the 'loadedmetadata' event never calls and hence videos not appear.

**Comment by ayaz-khan-niazi** [comment]
Same issue on my side. Does anyone fix it?

**Comment by budingts** [comment]
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://127.0.0.1:443/peerjs/peerjs/id?ts=16384355359790.35022730803790936. (Reason: CORS request did not succeed).

**Comment by budingts** [comment]
I got it
everyone show set port connect 


> Hi! When a new user joins the room, it looks like the `join-room` event is not emitted. I put a `console.log` on:
> 
> ```
> peer.on("open", (id) => {
>   socket.emit("join-room", ROOM_ID, id, user);
> });
> ```
> 
> And it does not reach this part. Thus, the `user-connected` event does not get broadcasted.
端口使用错误了 ，你应该设置正确的端口，设置peer的端口和node端口一致就行了

![image](https://user-images.githubusercontent.com/5056373/144391582-c12ddb69-0fbe-4cd2-9f53-d40a209dfbdd.png)
![image](https://user-images.githubusercontent.com/5056373/144391650-86306dbc-9bae-44cc-9be2-bc131cf35895.png)
都设置成3030，后者其他就行了

**Comment by ecreamer** [comment]
I had the same issue! I hope my solution helps you:

Inside of server.js, I replaced this line:

**socket.to(roomId).broadcast.emit("user-connected", userId);**

with:

**socket.on('ready', ()=>
    {
      socket.to(roomId).broadcast.emit("user-connected", userId);
    })**

Then, inside of script.js, immediately after this section:

**peer.on("call", (call) => {
      call.answer(stream);
      const video = document.createElement("video");
      call.on("stream", (userVideoStream) => {
        addVideoStream(video, userVideoStream);
      });
    });**

I added:

**socket.emit("ready")**

**Comment by Radiqa-20** [comment]
The joining of newUser issue can be resolve by adding this line in Socker.on function.
  socket.on("user-connected", (userId) => {
            console.log("User connected: " + userId);
            connectToNewUser(userId, myVideoStream);
        });
